server/api.go

```diff
@@ -341,6 +344,64 @@ func (a *API) disconnect(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (a *API) getConnectedChannels(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodOptions {
+		return
+	}
+
+	userID := r.Header.Get(HeaderMattermostUserID)
+	links, err := a.p.store.ListChannelLinksWithNames()
+	if err != nil {
+		a.p.API.LogError("Error occurred while getting the linked channels", "Error", err.Error())
+		http.Error(w, "Error occurred while getting the linked channels", http.StatusInternalServerError)
+		return
+	}
+
+	paginatedLinks := []*storemodels.ChannelLink{}
+	if len(links) > 0 {
+		sort.Slice(links, func(i, j int) bool {
+			return links[i].MattermostChannelID < links[j].MattermostChannelID
+		})
+
+		msTeamsTeamIDsVsNames, msTeamsChannelIDsVsNames, err := a.p.GetMSTeamsTeamAndChannelDetailsFromChannelLinks(links, userID, true)
+		if err != nil {
+			a.p.API.LogError("Unable to get the MS Teams channels and teams detail", "Error", err.Error())
+			http.Error(w, "Unable to get the MS Teams channels and teams detail", http.StatusInternalServerError)
+			return
+		}
+
+		offset, limit := a.p.GetOffsetAndLimitFromQueryParams(r)
+		for index, link := range links {
```

Comment:
can we directly start our loop from the offset index to avoid extra iterations and computations?